src/utils/internalLinks.js

```diff
@@ -173,13 +176,12 @@ export const getNarrowFromNarrowLink = (
   if (
     hashSegments.length === 2
     && hashSegments[0] === 'is'
-    && /^(private|starred|mentioned)/i.test(hashSegments[1])
+    && (hashSegments[1] === 'starred'
+      || hashSegments[1] === 'mentioned'
+      || hashSegments[1] === 'dm' // new in server-future; means the same as 'private'
```

Comment:
Instead of "server-future", I think we can just say "server-7.0". It's not a dead certainty this will land in server 7.0, but I think it's highly probable; it's not a feature with a lot of risk.